### DISCUSSION TITLE
HttpStatusException 예외 메세지 처리 논의

### DISCUSSION THREAD

**Discussion by kimhyun5u** [woowa-techcamp-2024/Team1-3K1K#30] [General]
논의 배경: 공통 Exception 을 정의하면서 기존 Controller 단위에서 사용하던 ErrorCode 가 애매해짐. 또한, 서버 로그와 클라이언트 에러 메시지를 다르게 설정하고 싶은 요구가 있음.

**Comment by Dr-KoKo** [comment]
24-08-13 회의 내용입니다.

### 서버 에러 메시지와 클라이언트 에러 메시지를 분리하고 싶다.
  - 어떤 방식으로 분리할 것인가
    1. Exception message 에 생성자로 넘겨주기 ✅
        - 클라이언트로의 응답과 서버 로그가 매칭이 되었으면 좋겠다.
    2. ErrorCode 에 서버 로그, 클라이언트 메시지를 정의해준다.
        - 단점: 동적으로 메세지를 처리할 수 없음

### 결론
- HttpStatusException에 HttpStatusException(ErrorCode errorCode,String message) 생성자를 만들자.
   - 기존의 생성자는 Deprecate
- Exception을 던질 때 서버로 남길 메시지를 생성자로 넘기자
